std/file.d

```diff
@@ -2219,15 +2219,31 @@ unittest
         $(D FileException) on error (which includes if the _symlink already
         exists).
   +/
-version(StdDdoc) void symlink(C1, C2)(const(C1)[] original, const(C2)[] link) @safe;
-else version(Posix) void symlink(C1, C2)(const(C1)[] original, const(C2)[] link) @safe
-{
-    static auto trustedSymlink(const(C1)[] path1, const(C2)[] path2) @trusted
+version(StdDdoc) void symlink(RO, RL)(RO original, RL link)
+    if ((isInputRange!RO && isSomeChar!(ElementType!RO) ||
+            isConvertibleToString!RO) &&
+        (isInputRange!RL && isSomeChar!(ElementType!RL) ||
+            isConvertibleToString!RL));
+else version(Posix) void symlink(RO, RL)(RO original, RL link)
+    if ((isInputRange!RO && isSomeChar!(ElementType!RO) ||
+            isConvertibleToString!RO) &&
+        (isInputRange!RL && isSomeChar!(ElementType!RL) ||
+            isConvertibleToString!RL))
+{
+    static if (isConvertibleToString!RO || isConvertibleToString!RL)
+    {
+        import std.meta : staticMap;
+        alias Types = staticMap!(convertToString, RO, RL);
+        symlink!Types(original, link);
+    }
+    else
     {
-        return core.sys.posix.unistd.symlink(path1.tempCString(),
-                                             path2.tempCString());
```

Comment:
It made sense that the calls to `tempCString` were within the `@trusted` block before, despite `tempCString` itself being mistakenly `@trusted`. But I guess the unit tests would ensure that this is properly rectified if we decide to fix `tempCString`.